agent/src/beerocks/slave/tasks/task.h

```diff
@@ -0,0 +1,70 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _TASK_H_
+#define _TASK_H_
+
+#include <beerocks/tlvf/beerocks_header.h>
+#include <tlvf/CmduMessageRx.h>
+
+namespace beerocks {
+
+class Task {
+public:
+    Task() {}
+    virtual ~Task() {}
+
+    /**
+     * @brief Enum which defines all tasks type.
+     * 
+     * Each task must have an entry on the enum eType, so it will be possible to send the task an
+     * event. 
+     */
+    enum eType : uint8_t {
+        TOPOLOGY,
+    };
+
+    /**
+     * @brief Retrun the task type.
+     * 
+     * Called by TaskPool send_event().
+     * 
+     * @return eType Task type.
+     */
+    virtual eType get_task_type() = 0;
+
+    /**
+     * @brief Work function. Shall do the task routine. Called by TaskPool run_task() function.
+     */
+    virtual void work(){};
+
+    /**
+     * @brief Handle events, which are being send from containing thread.
+     * 
+     * @param event_enum_value Event enum value which shall be defined on the task.
+     */
+    virtual void handle_event(uint8_t event_enum_value){};
```

Comment:
1. At his point I really feel that I would like to have a documentation explaining the idea of `Task` and what it should do and how it is designed.
2. It seems strange that the default implementation does nothing with the event.
```suggestion
    virtual void handle_event(uint8_t event_enum_value) = 0;
```

Reply:
This is because not all tasks will necessarily handle events.
It is optional.